test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/RuntimeTypeTestsJSTest.scala

```diff
@@ -0,0 +1,163 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.compiler
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSGlobal
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalajs.testsuite.utils.AssertThrows._
+import org.scalajs.testsuite.utils.Platform._
+
+class RuntimeTypeTestsJSTest {
+  import RuntimeTypeTestsJSTest._
+
+  // Import the helpers from the non-JS test
+  import RuntimeTypeTestsTest._
+
+  @Test def scalaArraysOfJSTypes(): Unit = {
+    val arrayOfParentJSType = new Array[ParentJSType](0)
```

Comment:
The name `ParentJSType` might be misleading here: The distinction between class / interface is crucial for some of the tests here. So we should probably call it `ParentJSInterface`?